lib/firmata.js

```diff
@@ -1411,6 +1459,178 @@ Board.prototype.stepperStep = function(deviceNum, direction, steps, speed, accel
 };
 
 /**
+ * Asks the Arduino to configure a hardware or serial port.
+ * @param {object} options Options:
+ *   portId {number} The serial port to use (HW_SERIAL1, HW_SERIAL2, HW_SERIAL3, SW_SERIAL0,
+ *   SW_SERIAL1, SW_SERIAL2, SW_SERIAL3)
+ *   baud {number} The baud rate of the serial port.
+ *   bytesToRead {number} The number of bytes to read on each iteration of the main loop
+ *   rxPin {number} [SoftwareSerial only] The RX pin of the SoftwareSerial instance
+ *   txPin {number} [SoftwareSerial only] The TX pin of the SoftwareSerial instance
+ */
+
+Board.prototype.serialConfig = function(options) {
+
+  var portId;
+  var baud;
+  var bytesToRead;
+  var rxPin;
+  var txPin;
+
+  if (typeof options === "object" && options !== null) {
+    portId = options.portId;
+    baud = options.baud;
+    bytesToRead = options.bytesToRead;
+    rxPin = options.rxPin;
+    txPin = options.txPin;
+  }
+
+  // TODO - if info about the Serial capabilities can be added to the capability query, a
+  // reasonable default value can be added and this error can be removed. Until then it is
+  // not possible to know if a board supports only HW serial or only SW serial or both so
+  // there is really no way to add a default portId value.
+  if (typeof portId === "undefined") {
+    throw new Error("portId must be specified, see SERIAL_PORT_IDs for options.");
+  }
+
+  baud = baud || 57600;
+  bytesToRead = bytesToRead || 0;
+
+  var data = [
+    START_SYSEX,
+    SERIAL_MESSAGE,
+    SERIAL_CONFIG | portId,
+    baud & 0x007F,
+    (baud >> 7) & 0x007F,
+    (baud >> 14) & 0x007F,
+    bytesToRead & 0x007F,
+    (bytesToRead >> 7) & 0x007F
+  ];
+  if (rxPin) {
+    data.push(rxPin);
+  }
+  if (txPin) {
+    data.push(txPin);
+  }
```

Comment:
Yeah, definitely enforce that for SoftwareSerial—throw an error exception, that will get the programmer's attention